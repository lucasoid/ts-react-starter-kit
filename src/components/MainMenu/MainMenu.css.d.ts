export const mainMenu: string;
export const mainMenuItem: string;
export const active: string;
