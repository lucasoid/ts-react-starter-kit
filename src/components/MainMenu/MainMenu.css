.mainMenu {
    margin: 0;
    padding: 0;
    list-style-type: none;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}

.mainMenu.light {
    background: #fff;
    color: #444;
    border-bottom: 1px solid #cecece;
    border-top: 1px solid #cecece;
}

.mainMenu.dark {
    background: #444;
    color: #fff;
    border-bottom: 1px solid #888;
    border-top: 1px solid #888;
}

.mainMenuItem a {
    padding: 0.5em 1em;
    display: block;
    text-decoration: none;
    color: inherit;
    border-bottom: 2px solid transparent;
}

.mainMenuItem a.active {
    border-bottom: 2px solid teal;
}

.mainMenu.light .mainMenuItem a:hover {
    background-color: #dfdfdf;
    border-bottom: 2px solid #ccc;
}

.mainMenu.dark .mainMenuItem a:hover {
    background-color: #202020;
    border-bottom: 2px solid #222;
}
