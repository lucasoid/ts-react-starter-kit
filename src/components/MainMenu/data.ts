export default [
    {
        label: 'Item 1',
        href: '/item1',
    },
    {
        label: 'Item 2',
        href: '/item2',
    },
    {
        label: 'Item 3',
        href: '/item3',
    },
];
