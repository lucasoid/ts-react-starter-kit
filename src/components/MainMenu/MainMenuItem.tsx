import * as React from 'react';
import * as styles from './MainMenu.css';

interface MainMenuItemProps {
    children?: any;
}

export const MainMenuItem: React.FC<MainMenuItemProps> = props => (
    <li className={styles.mainMenuItem}>{props.children}</li>
);
