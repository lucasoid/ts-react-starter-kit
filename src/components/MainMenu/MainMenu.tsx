import * as React from 'react';
import { NavLink } from 'react-router-dom';
import { MainMenuItem } from './MainMenuItem';
import * as styles from './MainMenu.css';
import { ThemeContext, Themes } from '~theme';

export const MainMenu: React.FC<{}> = () => {
    let { theme } = React.useContext(ThemeContext);

    return (
        <ul className={`${styles.mainMenu} ${theme === Themes.DARK ? styles.dark : styles.light}`}>
            <MainMenuItem>
                <NavLink to="/" exact={true} activeClassName={styles.active}>
                    Home
                </NavLink>
            </MainMenuItem>
            <MainMenuItem>
                <NavLink to="/settings" exact={true} activeClassName={styles.active}>
                    Settings
                </NavLink>
            </MainMenuItem>
            <MainMenuItem>
                <NavLink to="/About" exact={true} activeClassName={styles.active}>
                    About
                </NavLink>
            </MainMenuItem>
        </ul>
    );
};
