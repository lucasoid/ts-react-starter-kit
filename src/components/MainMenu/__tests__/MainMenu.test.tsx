import * as React from 'react';
import { shallow } from 'enzyme';
import { MainMenu } from '../MainMenu';
import { Themes } from '~theme';

jest.mock('react', () => {
    const original = jest.requireActual('react');
    return {
        __esModule: true,
        ...original,
        useContext: jest.fn(),
    };
});

describe('MainMenu', () => {
    it('displays a themed main menu UI', () => {
        // @ts-ignore
        React.useContext.mockImplementationOnce(() => ({
            theme: Themes.LIGHT,
        }));
        let lightWrapper = shallow(<MainMenu />);
        expect(lightWrapper).toMatchSnapshot();

        // @ts-ignore
        React.useContext.mockImplementationOnce(() => ({
            theme: Themes.DARK,
        }));
        let darkWrapper = shallow(<MainMenu />);
        expect(darkWrapper).toMatchSnapshot();
    });
});
