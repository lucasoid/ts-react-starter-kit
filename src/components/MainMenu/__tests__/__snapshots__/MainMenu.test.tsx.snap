// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu displays a themed main menu UI 1`] = `
<ul
  className="mainMenu light"
>
  <MainMenuItem>
    <NavLink
      activeClassName="active"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
  </MainMenuItem>
  <MainMenuItem>
    <NavLink
      activeClassName="active"
      exact={true}
      to="/settings"
    >
      Settings
    </NavLink>
  </MainMenuItem>
  <MainMenuItem>
    <NavLink
      activeClassName="active"
      exact={true}
      to="/About"
    >
      About
    </NavLink>
  </MainMenuItem>
</ul>
`;

exports[`MainMenu displays a themed main menu UI 2`] = `
<ul
  className="mainMenu dark"
>
  <MainMenuItem>
    <NavLink
      activeClassName="active"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
  </MainMenuItem>
  <MainMenuItem>
    <NavLink
      activeClassName="active"
      exact={true}
      to="/settings"
    >
      Settings
    </NavLink>
  </MainMenuItem>
  <MainMenuItem>
    <NavLink
      activeClassName="active"
      exact={true}
      to="/About"
    >
      About
    </NavLink>
  </MainMenuItem>
</ul>
`;
